build/ReleaseNotesBuilder/ReleaseNotesModel.cs

```diff
@@ -0,0 +1,225 @@
+        private readonly List<string> _frontBugs;
+        private readonly List<string> _frontSecurity;
+
+        private readonly string? _preamble;
+        private readonly List<string> _bodyNewFeatures;
+        private readonly List<string> _bodyFixes;
+        private readonly List<string> _bodySecurity;
+        private readonly List<string> _bodyNotice;
+        private readonly string _checksums;
+        private readonly string? _epilogue;
+
+        public ReleaseNotesModel(PersistentData persistentData, string checksums)
+        {
+            _frontFeatures = new List<string>();
+            _frontBugs = new List<string>();
+            _frontSecurity = new List<string>();
+            _bodyNewFeatures = new List<string>();
+            _bodyFixes = new List<string>();
+            _bodySecurity = new List<string>();
+            _bodyNotice = new List<string>();
+
+            _subject = persistentData.Subject;
+            _releaseDate = DateTime.Now.ToString("yyyy-MM-dd");
+            _downloadLink = persistentData.DownloadLink;
+            _preamble = persistentData.Preamble;
+            _epilogue = persistentData.Epilogue;
+            _checksums = checksums;
+        }
+
+        public string GetDocumentContents()
+        {
+            var problems = Validate();
+            if (problems.Any())
+            {
+                Program.ExitWithError(ExitCode.InvalidData,
+                    "The following problems occurred building the release notes:"
+                    + Environment.NewLine + "  "
+                    + string.Join(Environment.NewLine + "  ", problems)
+                    );
+            }
+
+            var builder = new StringBuilder();
+            builder.AppendLine(FrontMatterWrapper);
+            builder.AppendLine($"{FrontMatterSubject}: {_subject}");
+            builder.AppendLine($"{FrontMatterReleaseDate}: '{_releaseDate}'");
+            builder.AppendLine($"{FrontMatterVersion}: {_releaseVersion}");
+            builder.AppendLine($"{FrontMatterDownloadLink}: '{_downloadLink}'");
+            builder.AppendLine($"{FrontMatterFeatures}: [{string.Join(',', _frontFeatures)}]");
+            builder.AppendLine($"{FrontMatterBugs}: [{string.Join(',', _frontBugs)}]");
+            builder.AppendLine($"{FrontMatterSecurity}: [{string.Join(',', _frontSecurity)}]");
+            builder.AppendLine(FrontMatterWrapper);
+            builder.AppendLine();
+
+            if (!string.IsNullOrWhiteSpace(_preamble))
+            {
+                builder.AppendLine(_preamble);
+                builder.AppendLine();
+            }
+
+            if (_bodyNotice.Any())
+            {
+                builder.AppendLine(Program.NoticeSection);
+                builder.AppendLine();
+                foreach (var entry in _bodyNotice)
+                {
+                    builder.AppendLine(entry);
+                }
+
+                builder.AppendLine();
+            }
```

Comment:
This code is repeated multiple times in this method, and it might make sense to refactor it out into its own method to help ensure that each section is formatted consistently.

Reply:
I'll look into it.